File: code/java/剑指offer57.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Arrays.sort(nums);
        int i=0,j=nums.length-1;
        while(nums[i]+nums[j]!=target){
            if(nums[i]+nums[j]>target) j--;
            else i++;
        }
        return new int[]{nums[i],nums[j]};
    }
}

File: code/python/剑指offer57.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums = sorted(nums)
        i = 0
        j = len(nums)-1
        while nums[i]+nums[j]!=target:
            if nums[i]+nums[j]<target:
                i+=1
            else:
                j-=1
        return [nums[i],nums[j]]